contracts/TradetrustERC721.sol

```diff
@@ -36,15 +39,37 @@ contract TradeTrustERC721 is ERC721MintableFull, IERC721Receiver {
     require(ownerOf(_tokenId) == address(this), "Cannot send token: Token not owned by token registry");
     _safeTransferFrom(ownerOf(_tokenId), to, _tokenId, "");
   }
+}
 
+contract TradeTrustERC721WithTitleEscrow is TradeTrustERC721 {
+  event TitleEscrowDeployed(
+    address indexed escrowAddress,
+    address indexed tokenRegistry,
+    address beneficiary,
+    address holder
+  );
+  constructor(string memory name, string memory symbol) public TradeTrustERC721(name, symbol) {}
+
+  function deployNewTitleEscrow(address beneficiary, address holder) external returns (address) {
+    TitleEscrow newTitleEscrow = new TitleEscrow(this, beneficiary, holder);
+    emit TitleEscrowDeployed(address(newTitleEscrow), address(this), beneficiary, holder);
+    return address(newTitleEscrow);
+  }
+
+  function transferToNewTitleEscrow(
```

Comment:
Could you take a look at one of the test cases inside (token-registry > test > TradeTrustERC721.js) line 177. 
1) The {from: beneficiary1} is the same as msg.sender correct?
2) Some modifiers inside TitleEscrow.sol that were used by the old transferToTitleEscrow function couldnt be used in the new transferToNewTitleEscrow function, do i port them over too?

Reply:
hello RJ, i refactored the transferring of title escrow function and moved it into TradeTrustERC721 contract. Whenever you find time, could you help me check my implementation? Similarly for the tests. But im not sure if its the most ideal method to call _transferFrom directly instead of going through safeTransferFrom. But i believe there is sufficient validation before it gets to _transferFrom. If not, I will think of another implementation.